var/spack/repos/builtin/packages/mapnik/package.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Mapnik(AutotoolsPackage):
+    """
+    mapnik combines pixel-perfect image output with lightning-fast
+    cartographic algorithms, and exposes interfaces in C++, Python, and Node
+    """
+
+    homepage = "https://mapnik.org/"
+    url      = "https://github.com/mapnik/mapnik/releases/download/v3.0.22/mapnik-v3.0.22.tar.bz2"
+
+    version('3.0.22', sha256='930612ad9e604b6a29b9cea1bc1de85cf7cf2b2b8211f57ec8b6b94463128ab9')
+
+    depends_on('python', type=('build', 'run'))
+    # Build fails with boost@1.70
+    depends_on('boost@:1.69.0+regex+filesystem+system+icu+program_options+python cxxstd=11')
```

Comment:
All of these variants are required?

Reply:
Yep! (with the exception of python may be, which might be required by python-mapnik. I will double check. In that case, I can add a "python" variant to mapnik I guess.)

Reply:
You _should_ be able to do something like `depends_on('mapnik ^boost+python')` if you need to, no need for a variant.

Reply:
boost dependencies are indicated here https://github.com/mapnik/mapnik/blob/master/bootstrap.sh

Reply:
I tried what you suggested above about boost, I think it's not working because of the broken concretization
```
$ spack install py-python-mapnik
mapnik requires boost version :1.69.0, but spec asked for 1.70.0

$ spack install py-python-mapnik ^boost@1.69.0
==> Error: 1.70.0 does not satisfy 1.69.0
```

It is tempting to make boost an explicit dependency for py-mapnik although it might trigger a fresh install of mapnik (which takes a while by the way) if both packages aren't installed simulataneously.